Mathlib/Dynamics/Ergodic/Function.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2023 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Dynamics.Ergodic.Ergodic
+import Mathlib.MeasureTheory.Function.AEEqFun
+
+/-!
+# Functions invariant under (quasi)ergodic map
+
+In this file we prove that an a.e. strongly measurable function `g : α → X` that is a.e. invariant
+under a (quasi)ergodic is a.e. equal to a constant. We prove several versions of this statement with
+slightly different measurability assumptions. We also formulate a version for
+`MeasureTheory.AEEqFun` functions with all a.e. equalities replaced with equalities in the quotient
+space.
+-/
+
+open Function Set Filter MeasureTheory Topology TopologicalSpace
+
+variable {α X : Type _} [MeasurableSpace α] {μ : MeasureTheory.Measure α}
+
+/-- Let `f : α → α` be a (quasi)ergodic map. Let `g : α → X` is a null-measurable function from `α`
+to a nonempty space with a countable family of measurable sets separating points of a set `s` such
+that `f x ∈ s` for a.e. `x`. If `g` that is a.e.-invariant under `f`, then `g` is a.e. constant. -/
+theorem QuasiErgodic.ae_eq_const_of_ae_eq_comp_of_ae_range₀ [Nonempty X] [MeasurableSpace X]
+    {s : Set X} [HasCountableSeparatingOn X MeasurableSet s] (h : QuasiErgodic f μ)
+    (hs : ∀ᵐ x ∂μ, g x ∈ s) (hgm : NullMeasurable g μ) (hg_eq : g ∘ f =ᵐ[μ] g) :
+    ∃ c, g =ᵐ[μ] const α c := by
+  refine exists_eventuallyEq_const_of_eventually_mem_of_forall_separating MeasurableSet hs ?_
+  refine fun U hU ↦ h.ae_mem_or_ae_nmem₀ (s := g ⁻¹' U) (hgm hU) ?_b
+  refine (hg_eq.mono fun x hx ↦ ?_).set_eq
+  rw [← preimage_comp, mem_preimage, mem_preimage, hx]
+
+section CountableSeparatingOnUniv
+
+variable [Nonempty X] [MeasurableSpace X] [HasCountableSeparatingOn X MeasurableSet univ]
+  {f : α → α} {g : α → X}
+
+/-- If `f : α → α` is a (pre)ergodic map and `g : α → X` is a measurable function from `α` to a
+nonempty topological space with second countable topology that is invariant under `f`, then `g` is
```

Comment:
instead of `second countable topology`, could you put in this docstring something more in line with the statement? Same thing below.